Bugfix/daq move dataset

seb5g opened the pull request:
fixes #815 

The DAQ_Move created at instanciation an instance of a H5Saver to store DataActuator data when looping the get_actuator_value. But it's not needed at startup, it should be started only when looping.

Data were not even saved... now they are stored everytime one clicks the loop action

seb5g committed:
do not create an instance of module saver at startup but only when looping is activated!

and add data into the created h5saver

seb5g committed:
Update daq_move.py

Ashwolaa commented on packages/pymodaq/src/pymodaq/control_modules/daq_move.py:
```diff
@@ -20,6 +20,7 @@
 
 from easydict import EasyDict as edict
 
+from pymodaq.utils.h5modules.module_saving import ActuatorSaver
```
Why import if not used?

seb5g replied:
some automatic import added by pycharm I guess...

Ashwolaa reviewed (commented):
Looks good on usage, no extra dataset with DAQ_move. The update value seems to behave correctly

seb5g committed:
patched the use of h5saver to allow set settings before it is instanciated (otherwise) cannot not create a dataset!

there is a synchronisation between the saver settings replica in the main UI and the actual h5saver in the background

A widgetsync for trees would be handy for such a task

seb5g commented:
worked also on the daq_viewer dataset creation at startup. A small polish on the daq_move side still to do to replicate some work done on the viewer and we'll be ok.

@Ashwolaa do you think feasible a widgetsync for parameters that would sync their widget tree representation and their values that would ease a lot of implementation in pymodaq.

seb5g committed:
cleaning

Ashwolaa commented:
> worked also on the daq_viewer dataset creation at startup. A small polish on the daq_move side still to do to replicate some work done on the viewer and we'll be ok.
> 
> @Ashwolaa do you think feasible a widgetsync for parameters that would sync their widget tree representation and their values that would ease a lot of implementation in pymodaq.

I will look into it, I just PR an extension to dict base entries into the syncing mechanism so I am fairly optimistic but we should discuss about it.

seb5g committed:
similar implementation for daq_viewer/daq_move wrt continuous saving (logging for a single module)

seb5g commented:
@Ashwolaa the PR is finished and solved the issues. Since yesterday I incorporated a mechanisms to tell the DAQ_Move to log its values independently of the refresh state/button.

Could you check all this?

Ashwolaa commented:
When trying to launch a scan, I receive the error
ERROR:pymodaq.daq_scan:'DAQ_Viewer' object has no attribute 'module_and_data_saver'
Traceback (most recent call last):
  File "D:\Work\PyMoDAQ\packages\pymodaq\src\pymodaq\extensions\daq_scan.py", line 797, in set_scan
    res = self.update_scan_info()
  File "D:\Work\PyMoDAQ\packages\pymodaq\src\pymodaq\extensions\daq_scan.py", line 575, in update_scan_info
    scan_node = self.module_and_data_saver.get_set_node(new=False)
  File "D:\Work\PyMoDAQ\packages\pymodaq\src\pymodaq\utils\h5modules\module_saving.py", line 414, in get_set_node
    module.module_and_data_saver.main_module = False
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'DAQ_Viewer' object has no attribute 'module_and_data_saver'

Ashwolaa commented:
> @Ashwolaa the PR is finished and solved the issues. Since yesterday I incorporated a mechanisms to tell the DAQ_Move to log its values independently of the refresh state/button.
> 
> Could you check all this?

I am not sure to understand how it should work. When using the RefreshValue, nothing is being recorded. Should it appear in the C:\Data? aka the filepath given in config?

seb5g committed:
make sure no new h5 file are created when calling module_and_data_saver property

seb5g committed:
modify mock behaviour to reflect change in API

seb5g commented:
here a small use case. It is the same usage as continuous saving from the daq_viewer. Contrary to what I said above, the new implementation need you to tick `do_save` to trigger a "continuous saving", a king of single module logger. Then whenever get_actuator_value is called, the value is saved (works also with the refresh obviously) 


https://github.com/user-attachments/assets/9e1a50bf-8e0d-4926-a163-01726b3f3eaa

and I fixed the issue with the daq_scan and all the files that were also created with it. FOr future records, it is related to the way the property: module_and_data_saver is encoded (creating H5 files when calling the property and if no H5 has been set beforehand). This is what I do here by using not the property but the _module_and_data_saver variable.

I don't like too much this patch but I feel like the DAQ Scan would need a full cleanup in the future anyway...

Do you see other bugs?
